dask/array/random.py

```diff
@@ -17,7 +17,603 @@
+        seeds = random_rng_data(len(sizes), self._bit_generator_state)
+        token = tokenize(seeds, size, chunks, args, kwargs)
+        name = f"{funcname}-{token}"
+
+        keys = product(
+            [name], *([range(len(bd)) for bd in chunks] + [[0]] * len(extra_chunks))
+        )
+        blocks = product(*[range(len(bd)) for bd in chunks])
+
+        vals = []
+        for seed, size, slc, block in zip(seeds, sizes, slices, blocks):
+            arg = []
+            for i, ar in enumerate(args):
+                if i not in lookup:
+                    arg.append(ar)
+                else:
+                    if isinstance(ar, Array):
+                        arg.append((lookup[i],) + block)
+                    else:  # np.ndarray
+                        arg.append((getitem, lookup[i], slc))
+            kwrg = {}
+            for k, ar in kwargs.items():
+                if k not in lookup:
+                    kwrg[k] = ar
+                else:
+                    if isinstance(ar, Array):
+                        kwrg[k] = (lookup[k],) + block
+                    else:  # np.ndarray
+                        kwrg[k] = (getitem, lookup[k], slc)
+            vals.append(
+                (_apply_random_func, self._generator, funcname, seed, size, arg, kwrg)
+            )
+
+        meta = _apply_random_func(
+            self._generator,
+            funcname,
+            seed,
+            (0,) * len(size),
+            small_args,
+            small_kwargs,
+        )
+
+        dsk.update(dict(zip(keys, vals)))
+
+        graph = HighLevelGraph.from_collections(name, dsk, dependencies=dependencies)
+        return Array(graph, name, chunks + extra_chunks, meta=meta)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def beta(self, a, b, size=None, chunks="auto", **kwargs):
+        return self._wrap("beta", a, b, size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def binomial(self, n, p, size=None, chunks="auto", **kwargs):
+        return self._wrap("binomial", n, p, size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def chisquare(self, df, size=None, chunks="auto", **kwargs):
+        return self._wrap("chisquare", df, size=size, chunks=chunks, **kwargs)
+
+    with contextlib.suppress(AttributeError):
+
+        @derived_from(np.random.Generator, skipblocks=1)
+        def choice(
+            self,
+            a,
+            size=None,
+            replace=True,
+            p=None,
+            axis=0,
+            shuffle=True,
+            chunks="auto",
+        ):
+            dependencies = []
+            # Normalize and validate `a`
+            if isinstance(a, Integral):
+                # On windows the output dtype differs if p is provided or
+                # absent, see https://github.com/numpy/numpy/issues/9867
+                dummy_p = np.array([1]) if p is not None else p
```

Comment:
Shouldn't this be `is None`? Otherwise it seems the user's `p` value is always getting replace when specified 

```suggestion
                # On windows the output dtype differs if p is provided or
                # absent, see https://github.com/numpy/numpy/issues/9867
                dummy_p = np.array([1]) if p is None else p
```

Reply:
No, `dummy_p` is only used in the next line and only with a fixed `a` (so we cant use user supplied p) and is only needed because of the numpy bug mentioned in the comments.  That numpy bug was fixed in numpy-1.17.0.  Removed the comments and dummy_p